Add: custom js visualizations

luke14free opened the pull request:
![test](https://cloud.githubusercontent.com/assets/166602/20396510/b932ec56-ace6-11e6-9320-d027562333d8.gif)

arikfr commented on frontend/app/visualizations/chart/index.js:
```diff
@@ -75,6 +76,11 @@ function ChartEditor(ColorPalette) {
         });
       };
 
+      scope.options.customCode = "// Available variables are x, ys, element, and Plotly.\n";
```
Please use ES6 template strings here (created with backticks - "``"), which supports multi line strings.

arikfr commented on frontend/package.json:
```diff
@@ -53,7 +53,7 @@
     "ng-annotate-loader": "^0.2.0",
     "ng-table": "^2.1.0",
-    "plotly.js": "^1.16.0",
+    "plotly.js": "^1.2.0",
```
?

luke14free commented:
@arikfr I added the value to the clientConfig, but it was somehow defined as a constant in the `index.js` file. I guess that's it's linked to the fact that you want to separate backend and frontend and are still mocking that stuff, so I also added temporarily the switch to that constant in order to test the feature.

arikfr commented on frontend/app/visualizations/chart/index.js:
```diff
@@ -66,6 +65,11 @@ function ChartEditor(ColorPalette) {
       };
 
+      console.log(clientConfig);
+      if (clientConfig.allowCustomJSVisualizations) {
```
Need to use this in the render function too, so that someone clever won't manually submit this type.

arikfr commented:
@luke14free you understood it right about the `clientConfig` 👍

agenorfbp commented:
in my redash don't have many files listed in this pull, example: index.js.
I need create those files ?

agenorfbp commented:
This customization is very important for me and my organization, this Should be including in version 1.0
